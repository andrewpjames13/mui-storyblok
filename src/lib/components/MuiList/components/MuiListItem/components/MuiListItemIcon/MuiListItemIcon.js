import React from 'react';
import PropTypes from 'prop-types';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import Storyblok from '../../../../../../utils/Storyblok';
import MuiIcon from '../../../../../MuiIcon/MuiIcon';

/**
 * MuiListItemIcon is used in storyblok redirect to react routes
 */

export const MuiListItemIcon = ({
  content,
  rootClass,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <ListItemIcon className={styles.root}>
      <MuiIcon {...content[0]} />
    </ListItemIcon>
  );
};

export default MuiListItemIcon;

MuiListItemIcon.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Override or extend the styles applied to the component
   * */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /** MuiIcon Allowed maximum: 1 */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiListItemIcon.defaultProps = {
  rootClass: [],
};
