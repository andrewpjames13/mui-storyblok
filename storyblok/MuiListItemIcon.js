module.exports = {
  "name": "MuiListItemIcon",
  "display_name": null,
  "updated_at": "2020-03-31T18:27:05.478Z",
  "schema": {
    "rootClass": {
      "type": "options"
    },
    "content": {
      "type": "bloks",
      "maximum": "1",
      "restrict_components": true,
      "component_whitelist": [
        "MuiIcon"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiListItemIcon",
  "component_group_uuid": null
}