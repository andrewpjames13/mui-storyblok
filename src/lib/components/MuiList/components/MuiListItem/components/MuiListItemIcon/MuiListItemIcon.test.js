import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiListItemIcon from './MuiListItemIcon';

function setup() {
  const props = {
    content: [{
      component: 'MuiIcon',
      iconName: 'android',
    }],
  };
  const comp = shallow(<MuiListItemIcon {...props} />);
  return { comp, props };
}

describe('<MuiListItemIcon />', () => {
  it('renders MuiListItemIcon', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<MuiListItemIcon {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
