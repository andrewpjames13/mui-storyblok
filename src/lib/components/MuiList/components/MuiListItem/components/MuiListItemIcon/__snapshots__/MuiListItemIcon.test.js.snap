// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiListItemIcon /> snapshot 1`] = `
<div
  className="MuiListItemIcon-root makeStyles-root-3"
>
  <span
    aria-hidden={true}
    className="material-icons MuiIcon-root makeStyles-root-4 MuiIcon-colorSecondary"
  >
    android
  </span>
</div>
`;
